qualtran/bloqs/data_loading/qrom.py

```diff
@@ -157,7 +157,7 @@ def decompose_zero_selection(
                     ctrl=ctrl, junk=junk, target=and_target
                 )
             yield multi_controlled_and
-            yield self._load_nth_data(zero_indx, lambda q: CNOT().on(and_target, q), **target_regs)
+            yield self._load_nth_data(zero_indx, lambda q: cirq.CNOT(and_target, q), **target_regs)
```

Comment:
use qualtran `CNOT` so that it can immediately unwrap from CirqGateAsBloq without doing any logic?

Reply:
Using qualtran `CNOT` is slower because `CNOT` in Qualtran is not a `GateWithRegisters`. So the cirq construction wraps it in a `BloqAsCirqGate`; which ends up being slow. See the first profiling picture in my linked comment.